### PR TITLE
repos: add ListExternalRepoSpecs to only collect ExternalRepoSpec

### PR DESCRIPTION
**Pull request by unknwon**
Fetching all fields of all repos just for fields of `api.ExternalRepoSpec` is wasteful, so adds a dedicated method to do this job.

Part of #12699

### CODE REVIEW

File: cmd/repo-updater/repos/observability.go [hunk 1]
```diff
@@ -502,6 +502,27 @@ func (o *ObservedStore) ListRepos(ctx context.Context, args StoreListReposArgs)
...
 
+// ListExternalRepoSpecs calls into the inner Store and registers the observed results.
+func (o *ObservedStore) ListExternalRepoSpecs(ctx context.Context) (ids map[api.ExternalRepoSpec]struct{}, err error) {
+	tr, ctx := o.trace(ctx, "Store.ListExternalRepoSpecs")
+
+	defer func(began time.Time) {
+		secs := time.Since(began).Seconds()
+		count := float64(len(ids))
+
+		o.metrics.ListRepos.Observe(secs, count, &err)
```
Comment (asdine) [inline cmd/repo-updater/repos/observability.go:513]: Should it be `o.metrics.ListExternalRepoSpecs.Obsert(secs, count, &err)` instead?

**Review by asdine — APPROVED** [review]
Only one comment, approved to unblock

File: cmd/repo-updater/repos/observability.go (hunk 1 above)
Comment (unknwon) [reply cmd/repo-updater/repos/observability.go:513]: Good catch! 👀

Comment (unknwon) [reply cmd/repo-updater/repos/observability.go:513]: It turns out I miss all metrics for this new method :D

**Pull request merged**